src/hooks/useScrollable.ts

```diff
@@ -0,0 +1,58 @@
+import debounce from 'debounce';
+import { useCallback, useState } from 'react';
+
+export const useScrollable = () => {
```

Comment:
It simply checks whether scrolling is enabled or not. I tried using the modern method of using IntersectionObserver, but it behaved erratically on Android devices🥲